h2o-core/src/main/java/water/UDPClientEvent.java

```diff
@@ -26,22 +28,21 @@ AutoBuffer call(AutoBuffer ab) {
         // Connect event is not sent in multicast mode
         case CONNECT:
           if (H2O.isFlatfileEnabled()) {
-            Log.info("Client reported via broadcast message " + ce.clientNode + " from " + ab._h2o);
-
-            // It is important to propagate Client's HeartBeat information to the rest of the nodes
```

Comment:
Is this supposed to be removed?

Reply:
Not removed, just moved -> we do first intern and then report

Reply:
I am sorry, I do not see it there. Where is `setHeartBeat` being called?

Reply:
ah, this :) This is removed. The thought behind this is that the heartbeat will be filled when the node starts communication with the client

Reply:
I see. This means we are losing the information that a given node was a client until the heartbeat actually does come.

It might not be an issue, however, there are places that assume the information is available: like this one in Paxos:

```
      if(H2O.isFlatfileEnabled()){
        for(H2ONode n: H2O.getFlatfile()){
          if(!n._heartbeat._client && !PROPOSED.containsKey(n._key)){
```

Reply:
I see, I will revert this change, don't want to change any behaviour with this change. Thank you for great catch!

Reply:
I have put back the heart beat, please have a look @michalkurka